### PR TITLE
Added amqp exchange-type option.

### LINKED ISSUE #42: Support of different AMQP Exchange types
Currently only `direct` AMQP exchange type is supported. It is [hardcoded](https://github.com/spiral/jobs/blob/3a53d827eecc0b16b20f263fcb21e4c4edadff7f/broker/amqp/queue.go#L226) and not configurable.
Need support for other [exchange types](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchanges)

### PR DESCRIPTION
**Pull request by siad007**
Closes #42

### CODE REVIEW

**Comment by wolfy-j** [comment]
@48d90782 can you take a look please?

**Comment by 48d90782** [comment]
@wolfy-j Sure

File: broker/amqp/queue.go [hunk 1]
```diff
@@ -39,8 +41,16 @@ func newQueue(pipe *jobs.Pipeline, lsn func(event int, ctx interface{})) (*queue
 		return nil, fmt.Errorf("missing `queue` parameter on amqp pipeline")
 	}
 
+	isValidExchangeType := map[string]bool{"direct": true, "fanout": true, "topic": true, "headers": true}
+	exchangeType := pipe.String("exchange-type", "direct")
+
+	if !isValidExchangeType[exchangeType] {
+		return nil, fmt.Errorf("unknown `exchange-type` parameter on amqp pipeline")
```
Comment (48d90782) [inline broker/amqp/queue.go:48]: Error message should not contain backquotes.
```suggestion
		return nil, fmt.Errorf("unknown exchange type parameter on amqp pipeline")
```

File: broker/amqp/queue.go [hunk 2]
```diff
@@ -39,8 +41,16 @@ func newQueue(pipe *jobs.Pipeline, lsn func(event int, ctx interface{})) (*queue
 		return nil, fmt.Errorf("missing `queue` parameter on amqp pipeline")
 	}
 
+	isValidExchangeType := map[string]bool{"direct": true, "fanout": true, "topic": true, "headers": true}
+	exchangeType := pipe.String("exchange-type", "direct")
```
Comment (48d90782) [inline broker/amqp/queue.go:45]: I think it's better to have enum here for the exchange types. To be more obvious. For example in the `queue` struct:
`exchangeType    ExchangeType` where ExchangeType is:
```go
type ExchangeType string
```
```go
const (
	Direct  ExchangeType = "direct"
	Fanout  ExchangeType = "fanout"
	Topic   ExchangeType = "topic"
	Headers ExchangeType = "headers"
)
```
If needed, you can also implement `isValid` and `String` methods on this type.

**Comment by 48d90782** [comment]
Thank you @siad007

**Pull request merged**